test/GoTrueClient.test.ts

```diff
@@ -33,8 +34,12 @@ describe('GoTrueClient', () => {
       expect(session).not.toBeNull()
 
       await authWithSession.setSession(session?.refresh_token as string)
-      const { user } = await authWithSession.update({ data: { hello: 'world' } })
+      const {
+        data: { user },
+        error: updateError,
```

Comment:
We'll have to make a point in the docs of destructuring _after_ we check for the error. This will throw a JS error if `data` is `null` as it won't be able to destructure out the user. Wherever `{ data: { user }, error }` needs to be changed to:
```typescript
const { data, error } = authWithSession.xxx()

if (error) {
  throw error
}

const { user } = data
```

Reply:
Thanks - I think you've resolved every instance bar this one! (Unless GitHub is refusing to update because this part is marked for review or something?)

Reply:
fix in https://github.com/supabase/gotrue-js/pull/381